common/coconut-interface/src/lib.rs

```diff
@@ -79,27 +79,39 @@ impl VerifyCredentialBody {
     }
 }
 //  All strings are base58 encoded representations of structs
-#[derive(Serialize, Deserialize, Debug, Getters, CopyGetters)]
+#[derive(Clone, Serialize, Deserialize, Debug, Getters, CopyGetters)]
 pub struct BlindSignRequestBody {
     #[getset(get = "pub")]
     blind_sign_request: BlindSignRequest,
+    #[getset(get = "pub")]
+    tx_hash: String,
+    #[getset(get = "pub")]
+    signature: String,
     public_attributes: Vec<String>,
     #[getset(get = "pub")]
+    public_attributes_plain: Vec<String>,
+    #[getset(get = "pub")]
     total_params: u32,
 }
 
 impl BlindSignRequestBody {
     pub fn new(
         blind_sign_request: &BlindSignRequest,
+        tx_hash: String,
+        signature: String,
         public_attributes: &[Attribute],
+        public_attributes_plain: Vec<String>,
```

Comment:
Because we need to check the public attributes value against what's being recorded on the chain, especially the deposit value. And unless I'm missing something, there's no way to go back from `Attribute` to plain value, as it's basically a hash.